### PR TITLE
Expose classes and set nested type names

### PR DESCRIPTION
**Pull request by raghavsethi**
Required for non-JSON serialization.

### CODE REVIEW

**Comment by raghavsethi** [comment]
@martint: please take a look

File: event/src/main/java/io/airlift/event/client/EventFieldMetadata.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
 import static com.google.common.base.Objects.firstNonNull;
 import static io.airlift.event.client.EventDataType.validateFieldValueType;
 
-class EventFieldMetadata
+public class EventFieldMetadata
```
Comment (electrum) [inline event/src/main/java/io/airlift/event/client/EventFieldMetadata.java]: Annotate these with `@Beta`

File: event/src/main/java/io/airlift/event/client/EventTypeMetadata.java [hunk 2]
```diff
@@ -67,6 +67,11 @@
         return new EventTypeMetadata<T>(eventClass, Lists.<String>newArrayList(), Maps.<Class<?>, EventTypeMetadata<?>>newHashMap(), false);
     }
 
+    public static <T> EventTypeMetadata<T> getEventTypeMetadataNested(Class<T> eventClass)
```
Comment (electrum) [inline event/src/main/java/io/airlift/event/client/EventTypeMetadata.java]: Maybe call this `getNestedEventTypeMetadata`

Comment (raghavsethi) [inline event/src/main/java/io/airlift/event/client/EventTypeMetadata.java]: That method already exists and does something different.

File: event/src/main/java/io/airlift/event/client/EventFieldMetadata.java [hunk 3]
```diff
@@ -71,6 +73,21 @@ public String toString()
         this.containerType = containerType;
     }
 
+    public String getName()
+    {
+        return name;
+    }
+
+    public EventTypeMetadata<?> getNestedType()
```
Comment (martint) [inline event/src/main/java/io/airlift/event/client/EventFieldMetadata.java]: I'd make this return Optional, and add a comment in the javadoc explaining that it's "empty()" for atomic types.

**Comment by martint** [comment]
There are build failures

**Comment by raghavsethi** [comment]
Sorry about that - I didn't realize you were going to merge it. Just wanted you to check the other changes I made to have stuff be consistent after changing the field type to be Optional. Fixed the build failures.

**Comment by martint** [comment]
looks good

**Pull request merged**